xline/src/server/kv_server.rs

```diff
@@ -335,9 +344,34 @@ where
         Ok((puts, dels))
     }
 
-    /// Check if the current node is leader
-    fn is_leader(&self) -> bool {
-        self.state.is_leader()
+    /// Wait current node's state machine apply the conflict commands
+    async fn wait_read_state(&self, cmd: &Command) -> Result<(), tonic::Status> {
+        loop {
+            let rd_state = self
+                .client
+                .fetch_read_state(cmd)
+                .await
+                .map_err(|e| tonic::Status::internal(e.to_string()))?;
+            let wait_future = async move {
+                match rd_state {
+                    ReadState::Ids(ids) => {
+                        let fus = ids
+                            .into_iter()
+                            .map(|id| self.id_waiter.wait(id))
+                            .collect::<Vec<_>>();
+                        let _ignore = join_all(fus).await;
+                    }
+                    ReadState::CommitIndex(index) => {
+                        self.index_waiter.wait(index).await;
+                    }
+                    _ => unreachable!(),
+                }
+            };
+            if timeout(self.wait_synced_timeout, wait_future).await.is_ok() {
```

Comment:
Is it a good idea to make the kv_server mod aware of the "CurpConfig::wait_synced_timeout"?

Reply:
Added a new member to config